odoo/osv/query.py

```diff
@@ -98,6 +98,54 @@ def left_join(self, lhs_alias, lhs_column, rhs_table, rhs_column, link, extra=No
         return self._join('LEFT JOIN', lhs_alias, lhs_column, rhs_table, rhs_column, link, extra, extra_params)
 
     def _join(self, kind, lhs_alias, lhs_column, rhs_table, rhs_column, link, extra=None, extra_params=()):
+        """
+        Perform a join between a table already present in the current Query object and
+        another table.
+
+        :param str kind: the type of join to be added, e.g.::
+
+            query._join("LEFT JOIN", ...)
+
+        :param str lhs_alias: alias of a table already defined in the current Query object.
+        :param str lhs_column: column of `lhs_alias` to be used for the join's ON condition.
+        :param str rhs_table: name of the table to join to `lhs_alias`.
+        :param str rhs_column: column of `rhs_alias` to be used for the join's ON condition.
+        :param str link: used to generate the alias for the joined table, this string should
+            represent the relationship (the link) between both tables.
+        :param str extra: an sql string of a predicate or series of predicates to append to the
+            join's ON condition, `lhs_alias` and `rhs_alias` can be injected if the string uses
+            the `lhs` and `rhs` variables with the `str.format` syntax. e.g.::
+
+                query._join(..., extra="{lhs}.name != {rhs}.name OR ...", ...)
+
+        :param tuple extra_params: a tuple of values to be interpolated into `extra`, this is
+            done by psycopg2.
+
+        Full example:
+
+        >>> rhs_alias = query._join(
+        ...     "LEFT JOIN",
+        ...     "res_users",
+        ...     "partner_id",
+        ...     "res_partner",
+        ...     "id",
+        ...     "partner_id",           # partner_id is the "link" from res_users to res_partner
+        ...     "{lhs}.\"name\" != %s",
+        ...     ("Mitchell Admin",),
+        ... )
+        >>> rhs_alias
+        res_users_res_partner__partner_id
+
+        From the example above, the resulting query would be something like::
+
+            SELECT ...
+            FROM "res_users" AS "res_users"
+            LEFT JOIN "res_partner" AS "res_users_res_partner__partner_id"
+                ON "res_users"."partner_id" = "res_users_res_partner__partner_id"."id"
+                AND "res_users"."name" != 'Mitchell Admin'
+            WHERE ...
+
+        """
```

Comment:
This method is private and just serves as an implementation.  Maybe you could move the docstring to the public method `join` and refer to it in `left_join`.